config/log_config.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LoggerPatternConfig is an instance of a level specification for a given logger.
+type LoggerPatternConfig struct {
+	Pattern string `json:"pattern"`
+	Level   string `json:"level"`
+}
+
+const (
+	validLoggerSectionName             = `[a-zA-Z0-9]+([_-]*[a-zA-Z0-9]+)*`
+	validLoggerSectionNameWithWildcard = `(` + validLoggerSectionName + `|\*)`
+	validLoggerSections                = validLoggerSectionNameWithWildcard + `(\.` + validLoggerSectionNameWithWildcard + `)*`
+	validLoggerName                    = `^` + validLoggerSections + `$`
+)
+
+var loggerPatternRegexp = regexp.MustCompile(validLoggerName)
+
+func validatePattern(pattern string) bool {
+	return loggerPatternRegexp.MatchString(pattern)
+}
+
+// UpdateLoggerRegistry updates the logger registry if necessary  with the specified logConfig.
+func UpdateLoggerRegistry(logConfig []LoggerPatternConfig, loggerRegistry map[string]logging.Logger) (map[string]logging.Logger, error) {
+	newLogRegistry := make(map[string]logging.Logger)
+
+	for _, lpc := range logConfig {
+		if !validatePattern(lpc.Pattern) {
+			return nil, errors.New("Failed to Validate a Pattern")
```

Comment:
```suggestion
			return nil, errors.New("failed to Validate a Pattern")
```
The linter will likely complain, too, but errors in Go should start with lowercase.